hugr-core/src/hugr.rs

```diff
@@ -240,4 +303,158 @@ mod test {
         let hugr = simple_dfg_hugr();
         assert_matches!(hugr.get_io(hugr.root()), Some(_));
     }
+
+    const_extension_ids! {
+        const XA: ExtensionId = "EXT_A";
+        const XB: ExtensionId = "EXT_B";
+    }
+
+    #[rstest]
+    #[case([], XA.into())]
+    #[case([XA], XA.into())]
+    #[case([XB], ExtensionSet::from_iter([XA, XB]))]
+
+    fn infer_single_delta(
+        #[case] parent: impl IntoIterator<Item = ExtensionId>,
+        #[values(true, false)] remove: bool, // makes no difference when inferring
+        #[case] result: ExtensionSet,
+    ) {
+        let parent = ExtensionSet::from_iter(parent).union(TO_BE_INFERRED.into());
+        let (mut h, _) = build_ext_cfg(parent);
+        h.infer_extensions(remove).unwrap();
+        assert_eq!(h, build_ext_cfg(result).0);
+    }
+
+    #[test]
+    fn infer_removes_from_delta() {
+        let parent = ExtensionSet::from_iter([XA, XB]);
+        let mut h = build_ext_cfg(parent.clone()).0;
+        let backup = h.clone();
+        h.infer_extensions(false).unwrap();
+        assert_eq!(h, backup); // did nothing
+        h.infer_extensions(true).unwrap();
+        assert_eq!(h, build_ext_cfg(XA.into()).0);
+    }
+
+    #[test]
+    fn infer_bad_remove() {
+        let (mut h, mid) = build_ext_cfg(XB.into());
+        let backup = h.clone();
+        h.infer_extensions(false).unwrap();
+        assert_eq!(h, backup); // did nothing
+        let val_res = h.validate(&EMPTY_REG);
+        let expected_err = ExtensionError {
+            parent: h.root(),
+            parent_extensions: XB.into(),
+            child: mid,
+            child_extensions: XA.into(),
+        };
+        #[cfg(feature = "extension_inference")]
+        assert_eq!(
+            val_res,
+            Err(ValidationError::ExtensionError(expected_err.clone()))
+        );
+        #[cfg(not(feature = "extension_inference"))]
+        assert!(val_res.is_ok());
+
+        let inf_res = h.infer_extensions(true);
+        assert_eq!(inf_res, Err(expected_err));
+    }
+
+    fn build_ext_cfg(parent: ExtensionSet) -> (Hugr, Node) {
```

Comment:
```suggestion
    fn build_ext_dfg(parent: ExtensionSet) -> (Hugr, Node) {
```

?

Reply:
Erm, yes, oops, no idea how that happened